lib/roo/excelx/sheet.rb

```diff
@@ -113,6 +109,35 @@ def pad_cells(cell, last_column)
         (cell.coordinate.column - 1 - last_column).times { pad << nil }
         pad
       end
+
+      def first_last_row_col
+        @first_last_row_col ||= begin
+          first_row = last_row = first_col = last_col = nil
+
+          cells.each do |(row, col), cell|
+            if cell && !cell.empty?
```

Comment:
could be `next unless cell && !cell.empty?` instead of wrapping the whole thing in an if statement